libc/test/src/__support/freelist_heap_test.cpp

```diff
@@ -30,14 +30,19 @@ using LIBC_NAMESPACE::freelist_heap;
 #define TEST_FOR_EACH_ALLOCATOR(TestCase, BufferSize)                          \
   class LlvmLibcFreeListHeapTest##TestCase : public testing::Test {            \
   public:                                                                      \
+    FreeListHeapBuffer<BufferSize> fake_global_buffer;                         \
+    void SetUp() override {                                                    \
+      freelist_heap =                                                          \
+          new (&fake_global_buffer) FreeListHeapBuffer<BufferSize>;            \
+    }                                                                          \
```

Comment:
Since we have `heap_limit_fake.cpp`, is it be possible to define `_end` and `__llvm_libc_heap_limit` to be the start and end of some static global buffer so we can assert the linker-initialized `freelist_heap` is WAI?

Reply:
Not for these tests without limitation, since the buffer size is variable, but that does work for the malloc test, which gives us some smoke testing that the symbol version works. That's probably most of what's needed for now; the `morecore` variant should make this more straightfoward to test later. Done.